soql-sqlizer/src/main/scala/com/socrata/soql/sqlizer/ExprSqlizer.scala

```diff
@@ -40,8 +40,21 @@ class ExprSqlizer[MT <: MetaTypes with MetaTypesExt](
       } else {
         // We'll compress into a json array it because we don't want
         // ORDER BY's null positioning to be inconsistent depending on
-        // whether or not it's expanded
-        OrderBySql(expr.compressed, ascending = ascending, nullLast = nullLast)
+        // whether or not it's expanded.
+        //
+        // One wrinkle here: if expr is an expanded compound column
+        // which has been replaced with a select-list reference, we'll
+        // have to undo that replacement because we can't generate
+        //    compress(1, 2)
+        // and have it work.
+        val effectiveExpr =
+          (expr, expr.expr) match {
```

Comment:
This line hurts my brain a little but otherwise this all seems reasonable

Reply:
expr expr expr expr expr